src/mongo/db/ldap/ldap_manager_impl.cpp

```diff
@@ -614,12 +620,12 @@ Status LDAPManagerImpl::execQuery(std::string& ldapurl, std::vector<std::string>
         } else {
             BerElement *ber = nullptr;
             auto attribute = ldap_first_attribute(ldap, entry, &ber);
-            ON_BLOCK_EXIT([&] { ber_free(ber, 0); });
+            ON_BLOCK_EXIT([=] { ber_free(ber, 0); });
```

Comment:
From my perspective, capturing everything by value may have undesired side effects, e.g. accidental copying of a large object which isn't going to be used inside the lambda. I'd understand the "capture all" approach if the lambda required more than 80% of the automatic variables from the reachable scope _and_ that scope is quite small (e.g. fits in half of a typical screen height) so we can easily ensure no unwanted copying is done. However, we have the opposite situation in the `execQuery` function: each lambda uses only a few variables but the reachable scope (the `execQuery` function itself) is almost two screens long. That makes the "capture all" approach quite risky in this context.